File: 2002/1/9-1.rb
#!/usr/bin/env ruby

STDIN.each do |line|
  a,b,c,n = line.split(',').collect { |i| i.to_i }
  total = a + b + c
  (n-3).times do
    d = a + b + c
    total += d
    a,b,c = b,c,d
  end
  puts "The answer is: #{total}"
end

File: 2002/1/9-1.py
#!/usr/bin/env python

import sys

for line in sys.stdin.readlines():
    a,b,c,n = map(int, line.split(','))
    total = a + b + c
    for i in xrange(3,n):
        d = a + b + c
        total += d
        a, b, c = b, c, d
    print "The answer is: %s" % total
